Azure.ResourceManager.Core/SubscriptionOperations.cs

```diff
@@ -90,5 +74,10 @@ public async override Task<ArmResponse<Subscription>> GetAsync(CancellationToken
         {
             return s => new Subscription(ClientOptions, new SubscriptionData(s));
         }
+
+        protected internal override ResourceType GetValidResourceType()
```

Comment:
since this is auto-generated i wasn't worried about needing to override this everywhere, and this is a bit cleaner having each class implement their own behavior vs passing down all the necessary flags to implement if/else logic in the base class